nutils/function.py

```diff
@@ -3052,24 +3053,17 @@ def wrapper( *args, **kwargs ):
         ba.arguments[param.name] = param.default
 
     # replace argument `arg_key` with a derivative helper
-    if isinstance(arg_key, int):
-      args = list(ba.args)
-      orig = args[arg_key]
-      var = DerivativeHelper( orig.shape, arg_axes )
-      args[arg_key] = var
-      kwargs = ba.kwargs
-    elif arg_key in ba.kwargs:
-      args = ba.args
-      kwargs = dict(ba.kwargs)
-      orig = ba.kwargs[arg_key]
-      var = DerivativeHelper( orig.shape, arg_axes )
-      kwargs[arg_key] = var
-    else:
-      raise ValueError( 'argument not found: {!r}'.format( arg_key ) )
+    args = list(ba.args)
+    kwargs = dict(ba.kwargs)
+    keyargs = args if isinstance(arg_key,int) else kwargs
+    orig = keyargs[arg_key]
+    orig_axes = arg_axes if arg_axes is not None else tuple(range(orig.ndim))
```

Comment:
Small remark. I guess I would change `orig_axes = ...` into `arg_axes = ...`, e.g.

    if arg_axes is None:
      arg_axes = tuple(range(orig.ndim))

Not that `arg_axes` is well chosen, but `orig_axes` is just another name for the same thing. Then, you could revert the change on the last line.